framework/test/unit/modules/nft/method.spec.ts

```diff
@@ -760,4 +741,287 @@ describe('NFTMethod', () => {
+			checkEventResult<TransferEventData>(
+				methodContext.eventQueue,
+				1,
+				TransferEvent,
+				0,
+				{
+					senderAddress,
+					recipientAddress,
+					nftID: existingNFT.nftID,
+				},
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+		});
+
+		it('should throw and emit error transfer event if nft is locked', async () => {
+			await expect(
+				method.transfer(
+					methodContext,
+					lockedExistingNFT.owner,
+					recipientAddress,
+					lockedExistingNFT.nftID,
+				),
+			).rejects.toThrow('Locked NFTs cannot be transferred');
+			checkEventResult<TransferEventData>(
+				methodContext.eventQueue,
+				1,
+				TransferEvent,
+				0,
+				{
+					senderAddress: lockedExistingNFT.owner,
+					recipientAddress,
+					nftID: lockedExistingNFT.nftID,
+				},
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+		});
+
+		it('should resolve if all params are valid', async () => {
+			jest.spyOn(internalMethod, 'transferInternal');
+
+			await expect(
+				method.transfer(methodContext, existingNFT.owner, recipientAddress, existingNFT.nftID),
+			).resolves.toBeUndefined();
+			expect(internalMethod['transferInternal']).toHaveBeenCalledWith(
+				methodContext,
+				recipientAddress,
+				existingNFT.nftID,
+			);
+		});
+	});
+
+	describe('transferCrossChain', () => {
+		const senderAddress = utils.getRandomBytes(LENGTH_ADDRESS);
+		const recipientAddress = utils.getRandomBytes(LENGTH_ADDRESS);
+		const receivingChainID = utils.getRandomBytes(LENGTH_CHAIN_ID);
+		const messageFee = BigInt(1000);
+		const data = '';
+		const includeAttributes = false;
+
+		it('should throw and emit error transfer cross chain event if nft does not exist', async () => {
+			await expect(
+				method.transferCrossChain(
+					methodContext,
+					senderAddress,
+					recipientAddress,
+					nftID,
+					receivingChainID,
+					messageFee,
+					data,
+					includeAttributes,
+				),
+			).rejects.toThrow('NFT substore entry does not exist');
+			checkEventResult<TransferCrossChainEventData>(
+				methodContext.eventQueue,
+				1,
+				TransferCrossChainEvent,
+				0,
+				{
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+		});
+
+		it('should throw and emit error transfer cross chain event if nft is escrowed', async () => {
+			await expect(
+				method.transferCrossChain(
+					methodContext,
+					senderAddress,
+					recipientAddress,
+					escrowedNFT.nftID,
+					receivingChainID,
+					messageFee,
+					data,
+					includeAttributes,
+				),
+			).rejects.toThrow('NFT is escrowed to another chain');
+			checkEventResult<TransferCrossChainEventData>(
+				methodContext.eventQueue,
+				1,
+				TransferCrossChainEvent,
+				0,
+				{
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID: escrowedNFT.nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+		});
+
```

Comment:
Similarly as in comment for `method.ts`, it seems to me that the unit test for the case when `chainID` is equal to either the `OWN_CHAIN_ID` or `receivingChainID` is missing, see [this part](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1851-L1853) of the `transferCrossChain` verification in the LIP.